app/main.py

```diff
@@ -1,3 +1,56 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, int, float]) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Union[Distance, int, float]) -> Distance:
+        if not isinstance(other, Distance):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(self, other: (int, float)) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: (int, float)) -> Distance:
+        return Distance(round((self.km / other), 2))
```

Comment:
The `__truediv__` method should ensure that `other` is an instance of `int` or `float` to avoid division errors. Additionally, consider handling division by zero.